packages/story-utils/src/generateSnapshot.ts

```diff
@@ -0,0 +1,46 @@
+import { render, RenderResult } from '@testing-library/react';
+import { getStoriesFromStoryFileExports } from './getStories';
+import type { StoryData, StoryFileExports } from './getStories';
+import prepareStory from './prepareStory';
+
+type TestOptions<Args> = {
+  /**
+   * get the snapshot either synchronously, returning the ChildNode,
+   * or asynchronously returning a promise that resolves to the ChildNode.
+   */
+  getSnapshot?: (
+    wrapper: RenderResult,
+  ) => Promise<ChildNode | null> | ChildNode | null;
+  /**
+   * Override the args for the story. Useful for event handlers that may be missing.
+   */
+  argOverrides?: Partial<Args>;
+};
+
+function getDefaultSnapshot(wrapper: RenderResult) {
+  return wrapper.container.firstChild;
+}
+
+/**
+ * Runs snapshot tests on all stories imported with `import * as anthology from 'storypath.stories.tsx'`
+ * @param storyFileExports the exports of a .stories.tsx file including the default export with additional context
+ * @param options the options for the test to add necessary context specific to snapshot tests
+ */
+export default function generateSnapshots<
+  S extends StoryFileExports<Args>,
+  Args
+>(
+  storyFileExports: S,
+  { getSnapshot = getDefaultSnapshot, argOverrides }: TestOptions<Args> = {},
+): void {
+  const stories = getStoriesFromStoryFileExports(storyFileExports);
+
+  for (const [storyName, story] of Object.entries<StoryData>(stories)) {
+    if (story.parameters?.snapshot?.disabled) continue;
```

Comment:
Not necessary now (especially since this is a port from Along), but at some point we should skip disabled tests, so that test runs display how many were skipped.

```js
const testFn = story.parameters?.snapshot?.disabled ? test.skip : test;

testFn(`${storyName} is cool`, () => {
  // ...
});
```

We used to do something similar in axe-storybook-testing: https://github.com/chanzuckerberg/axe-storybook-testing/blob/894e39a5bf4f96c791d5aefa486d9ff568002ec6/src/Suite.ts#L31